src/ripple/app/ledger/Ledger.cpp

```diff
@@ -334,7 +334,7 @@ bool Ledger::addSLE (SLE const& sle)
 bool Ledger::addTransaction (uint256 const& txID, const Serializer& txn)
 {
     // low-level - just add to table
-    auto item = std::make_shared<SHAMapItem> (txID, txn.peekData ());
+    auto item = std::make_shared<SHAMapItem const> (txID, txn.peekData ());
```

Comment:
make `auto const&` and save a `shared_ptr` construction? Of course, this assumes `mTransactionMap` doesn't change things out from under us, but you'd know that better than me.